cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -206,9 +206,15 @@ public synchronized void create(final NamespaceMeta metadata) throws Exception {
 
       // if needed, run master environment specific logic
       MasterEnvironment masterEnv = MasterEnvironments.getMasterEnvironment();
-      if (cConf.getBoolean(Constants.Namespace.NAMESPACE_CREATION_HOOK_ENABLED)
-          && masterEnv != null) {
-        masterEnv.onNamespaceCreation(namespace.getNamespace(), metadata.getConfig().getConfigs());
+      if (masterEnv != null) {
+        // if namespace creation hook is not enabled, identity will be created in default namespace.
+        String namespaceName = NamespaceId.DEFAULT.getNamespace();
+        if (cConf.getBoolean(Constants.Namespace.NAMESPACE_CREATION_HOOK_ENABLED)) {
+          masterEnv.onNamespaceCreation(namespace.getNamespace(),
+              metadata.getConfig().getConfigs());
+          namespaceName = namespace.getNamespace();
```

Comment:
Should we use the default service account in the created namespace if namespace creation is enabled?

Reply:
makes sense, added the logic to skip creation of identity if namespace creation hook is enabled.